src/app/stack/card-details/card-details.component.ts

```diff
@@ -981,9 +1320,62 @@ export class CardDetailsComponent implements OnInit, OnChanges {
         return headers;
     }
 
+    private calBadge(report: MReportInformation, name: string): string {
```

Comment:
Can you be more descriptive about a function name?

Reply:
fixed